internal/engine/watchmanager.go

```diff
@@ -195,16 +195,27 @@ func (w *WatchManager) OnChange(ctx context.Context, st store.RStore) {
 			continue
 		}
 
+		filter, err := ignore.CreateFileChangeFilter(target)
+		if err != nil {
+			st.Dispatch(NewErrorAction(err))
+			return
+		}
+		tiltIgnoreFilter, err := dockerignore.DockerIgnoreTesterFromContents(tiltRoot, w.tiltIgnoreContents)
+		if err != nil {
+			st.Dispatch(NewErrorAction(err))
+		}
+		filter = model.NewCompositeMatcher([]model.PathMatcher{filter, tiltIgnoreFilter})
+
 		for _, d := range target.Dependencies() {
-			err = watcher.Add(d)
+			err = watcher.Add(d, filter)
```

Comment:
would it make more sense to pass `filter` in to fsWatcherMaker?

Reply:
Yes, I like that approach better.